enterprise/server/workflow/service/service.go

```diff
@@ -843,31 +849,40 @@ func (ws *workflowService) startWorkflow(webhookID string, r *http.Request) erro
 }
 
 // starts a CI runner execution and returns the execution ID.
-func (ws *workflowService) executeWorkflow(ctx context.Context, key *tables.APIKey, wf *tables.Workflow, wd *interfaces.WebhookData, workflowAction *config.Action, extraCIRunnerArgs []string) (string, error) {
+func (ws *workflowService) executeWorkflow(ctx context.Context, key *tables.APIKey, wf *tables.Workflow, wd *interfaces.WebhookData, workflowAction *config.Action, invocationID string, extraCIRunnerArgs []string) (string, error) {
 	ctx = ws.env.GetAuthenticator().AuthContextFromAPIKey(ctx, key.Value)
 	ctx, err := prefix.AttachUserPrefixToContext(ctx, ws.env)
 	if err != nil {
 		return "", err
 	}
 	in := instanceName(wf, wd, workflowAction.Name)
-	ad, err := ws.createActionForWorkflow(ctx, wf, wd, key, in, workflowAction, extraCIRunnerArgs)
+	ad, err := ws.createActionForWorkflow(ctx, wf, wd, key, in, workflowAction, invocationID, extraCIRunnerArgs)
+	if err != nil {
+		return "", err
+	}
+
+	execCtx, err := bazel_request.WithRequestMetadata(ctx, &repb.RequestMetadata{ToolInvocationId: invocationID})
 	if err != nil {
 		return "", err
 	}
 
-	executionID, err := ws.env.GetRemoteExecutionService().Dispatch(ctx, &repb.ExecuteRequest{
+	opStream, err := ws.env.GetRemoteExecutionClient().Execute(execCtx, &repb.ExecuteRequest{
```

Comment:
does this stream need to be closed or will it get closed automatically?

Reply:
Unfortunately since it's a one-directional receive stream it's not possible to close it from the client so it will hang around until the invocation is complete. This doesn't seem like something we need to worry about though, what do you think?

Reply:
Would it work to cancel the execCtx once we've recv'd the first msg? If not, seems OK to leave it running in the background.

Reply:
Oh that's a great idea. Completely forgot that was an option.